server/sonar-db-dao/src/test/java/org/sonar/db/qualityprofile/ActiveRuleDaoTest.java

```diff
@@ -319,21 +326,35 @@ public void delete() throws Exception {
       .setInheritance(INHERITED)
       .setCreatedAt(1000L)
       .setUpdatedAt(2000L);
-    underTest.insert(dbTester.getSession(), activeRule);
-    dbSession.commit();
+    underTest.insert(dbSession, activeRule);
 
     underTest.delete(dbSession, activeRule.getKey());
-    dbSession.commit();
 
     assertThat(underTest.selectByKey(dbSession,  ActiveRuleKey.of(profile1.getKey(), rule1.getKey()))).isAbsent();
   }
 
   @Test
-  public void does_not_fail_when_active_rule_does_not_exist() throws Exception {
+  public void delete_does_not_fail_when_active_rule_does_not_exist() throws Exception {
     underTest.delete(dbSession, ActiveRuleKey.of(profile1.getKey(), rule1.getKey()));
   }
 
   @Test
+  public void deleteByKeys_deletes_rows_from_table() throws Exception {
+    underTest.insert(dbSession, newRow(profile1, rule1));
+    underTest.insert(dbSession, newRow(profile1, rule2));
+    underTest.insert(dbSession, newRow(profile2, rule1));
+
+    underTest.deleteByProfileKeys(dbSession, asList(profile1.getKey(), "does_not_exist"));
```

Comment:
To check that `"does not exist"` does not yield errors should not be part of this test. If it is important, it should be a dedicated test, otherwise it should be removed.